src/main/java/com/exadel/discountwebapp/location/repository/LocationRepository.java

```diff
@@ -13,4 +13,6 @@
     List<Location> findAllByCountry_CountryCode(String countryCode);
 
     List<Location> findAllByCity(String city);
+
+    Location findByCountry_CountryCodeAndCityAndAddressLine(String countryCode, String city, String addressLine);
```

Comment:
For now this line can potentially return several Locations with the same countryCode - city - addressLine.
Let's consider adding unique composite index for (countryCode - city - addressLine) to the location

Reply:
Added, but commented

Reply:
In future we need somehow, as u said find duplicates and remove them